src/System.Private.CoreLib/shared/System/Reflection/MemberInfo.cs

```diff
@@ -38,7 +38,7 @@ public virtual Module Module
 
         public virtual IEnumerable<CustomAttributeData> CustomAttributes => GetCustomAttributesData();
         public virtual IList<CustomAttributeData> GetCustomAttributesData() { throw NotImplemented.ByDesign; }
-
+        public virtual bool IsCollectible => DeclaringType.IsCollectible;
```

Comment:
`DeclaringType` can be `null`

Reply:
Also, this may be not right for generic methods.

I think it would be better to just return `true` here.

Reply:
@janvorli, what should be the behavior if the declaring type is null?
```
public virtual void IsCollectible => DeclaringType?.IsCollectible ?? false;
```
?

Reply:
Wouldn't the implementation I added in [RuntimeMethodInfo.cs](https://github.com/dotnet/coreclr/blob/3460503d0ab6870a42d38ff4fd45fec74730f6d0/src/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.cs#L532) be called for a generic method?

Reply:
MethodInfo and MemberInfo are not sealed. People can create their own implementations.